(WIP) Add .mailmap file

jawnsy opened the pull request:
**DO NOT MERGE** This needs to be manually reviewed and modified before it's suitable for inclusion, particularly to determine the "correct" (most current) attribution and to ensure that we don't combine contributors inaccurately.

**What this PR does / why we need it**:

Adding a mailmap file allows for contributions to be attributed correctly when a person has committed with multiple email addresses.  The [git documentation](https://www.kernel.org/pub/software/scm/git/docs/git-shortlog.html#_mapping_authors) probably provides the best explanation of why this would be helpful, but it should also help with analytics such as Mirantis' [stackalytics](http://stackalytics.com/) site.

This was automatically generated by https://github.com/greenrd/genmailmap,
then manually reviewed and modified.

**Blame**: this was @philips' idea, see https://twitter.com/BrandonPhilips/status/781260219414216704 :smile:

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33713)

<!-- Reviewable:end -->

luxas commented:
What states who should be in this list?
Is it all kubernetes maintainers?

gregkh commented:
It should map all contributors that aren't "easy" to determine who they work for due to use of email addresses that are not trackable, or to combine different email addresses used for the same person, if they want this information known.  No need to have it list only maintainers.

philips commented:
We should probably maintain this as a separate repo at github.com/kubernetes/mailcap or github.com/kubernetes/community. This map isn't just for kubernetes.git. cc @sarahnovotny

Commit:
Add .mailmap file

This was created manually using the output of `git shortlog`, sorting by
name.  The "primary email address" was selected based on whichever address
appeared to be a work address, and in the case of ties, LinkedIn information
was used to determine the most recent employer.

jawnsy commented:
I discarded what we had before, and instead went through the list of maintainers manually, looking for duplicates. It's entirely possible that I missed some.

It's mostly cut-and-dried in the cases where a Gmail address and work address are used, but I'm not quite sure how to attribute maintainers like Brendan Burns (was at Google, now at Microsoft) or Jess Frazelle (was at Mesosphere, now at Google). Perhaps the best way to proceed in those cases is to ping each maintainer by adding a comment to the appropriate line in the diff?
